Feature/ong4 35/formulario edicion creacion de miembros

SoraBeat opened the pull request:
En esta rama se requeria un formulario con los campos: name, image, description y links de redes sociales. Estos estan validados con yup. El manejo de la descripcion esta realizada con CKEditor 5.  Al estar todos los campos validados y apretar en enviar se muestran todos los campos ingresados por consola.
**VER EN EL CODIGO**
Hay que envolver el formulario con FormikProvider y el componente CKEditor con Field ya que no contiene el atributo "name" para la validacion del mismo.
![image](https://user-images.githubusercontent.com/30334118/198845835-46adf3a0-393d-45ec-ae1e-12bbf0bee0b9.png)

SoraBeat committed:
Inicializado de rama e importacion de componentes para el formulario

SoraBeat committed:
Armado del formulario y redireccion con router

Validaciones de los campos, integracion dentro del componente, etc.

SoraBeat committed:
Fix prop-types y validacion de URL

SoraBeat committed:
Incorporacion de descripcion con CKEditor

milandre commented on src/Components/Members/MembersForm.jsx:
```diff
@@ -0,0 +1,126 @@
+import Form from "../common/Form/Form";
+import FormTitle from "../common/Form/FormTitle";
+import InputImage from "../common/Form/InputImage";
+import FormError from "../common/Form/FormError";
```
Sugerencia:
Para las importaciones de este estilo que se hacen largas (../../../../), es mejor configurar rutas absolutas. Se puede hacer mediante un jsconfig.json que se crea en el directorio base de tu proyecto (al nivel del package.json) y hay una opción donde puedes elegir tu directorio base y hacer las importaciones a partir de ahi.

Ejemplo:

jsconfig.json
{ "compilerOptions": { "baseUrl": "src" }, "include": ["src"] }

Quieres importar en un archivo un componente que está en src/components/Button.js, puedes hacerlo asi:
import Button from 'components/Button';

Más información en: https://create-react-app.dev/docs/importing-a-component/#absolute-imports

milandre commented on src/Components/Members/MembersForm.jsx:
```diff
@@ -0,0 +1,126 @@
+		description: yup.string().required("Descripcion obligatoria"),
+		facebookUrl: yup
+			.string()
+
```
linea extra de espacio que debe borrarse

milandre commented on src/Components/Members/MembersForm.jsx:
```diff
@@ -0,0 +1,126 @@
+			.required("Link obligatorio"),
+		linkedinUrl: yup
+			.string()
+
```
linea extra de espacio que debe borrarse

milandre commented on src/Components/Members/MembersForm.jsx:
```diff
@@ -0,0 +1,126 @@
+			.string()
+			.min(4, "Minimo 4 caracteres")
+			.required("Nombre obligatorio"),
+		image: yup.string().required("Imagen obligatoria"),
```
Falta validación para que solo sean imagenes .png y ,jpg

milandre commented on src/Components/Members/MembersForm.jsx:
```diff
@@ -0,0 +1,126 @@
+	return (
+		<FormikProvider value={formik}>
+			<Form handleSubmit={handleSubmit}>
+				<FormTitle>Crear/Editar Miembro</FormTitle>
```
Esto debería mostrar un Editar Miembro cuando estas en la vista de edición y un Crear Miembro cuando estás en la vista de creación

milandre commented on src/Components/Members/MembersForm.jsx:
```diff
@@ -0,0 +1,126 @@
+					)}
+				</Field>
+				<FormError error={errors.description} touched={touched.description} />
+				<FormSubmitButton>Enviar</FormSubmitButton>
```
Dejar un poco más de espacio entre el componente FormError y el botón de FormSubmitButton. 
Se ve así actualmente:
![image](https://user-images.githubusercontent.com/5572252/199382152-8288afdb-a67e-4c28-b26e-d21d1c1477a7.png)

milandre replied:
no vi que esto esté con un margen. No es bloqueante pero estaría bueno agregarlo

milandre reviewed (changes requested):
@SoraBeat además de los reviews que hice, acá también faltaría incluir los cambios con respecto al componente de InputImage que te hice observación en el PR de ONG4 33 https://github.com/alkemyTech/React-SXXI-T4/pull/1#discussion_r1011044773

SoraBeat committed:
Cambios del pr

-Rutas  absolutas
-Lineas extras en blanco
-Validacion imagen yup
-Nombre del formulario segun  si esta editando o creando
-Margen superior boton FormSubmitButton

SoraBeat committed:
Sacar dos de los formatos adminitos en imagen

SoraBeat committed:
Cambios en el formulario pr

-Validacion de imagen
-Espacios inecesarios
-Ojito boton contraseña

SoraBeat committed:
Borrado de ruta inecesaria

SoraBeat committed:
Mover la declaracion de formatos soportados

milandre commented on src/Components/Members/MembersForm.jsx:
```diff
@@ -0,0 +1,130 @@
+			.string()
+			.min(4, "Minimo 4 caracteres")
+			.required("Nombre obligatorio"),
+		image: yup
```
Acá faltaría verificar de igual forma la validación, como lo comentado en #1

SoraBeat replied:
Cambios realizados en el commit [1e5a828] "Validacion de imagen arreglada"

milandre commented on src/Components/Members/MembersForm.jsx:
```diff
@@ -0,0 +1,130 @@
+		description: "",
+	};
+
+	const SUPPORTED_FORMATS = [
```
Acá faltaría verificar de igual forma la validación, como lo comentado en #1

SoraBeat replied:
Cambios realizados en el commit [b246e0e] "Mover declaracion de formatos soportados"

milandre replied:
Tengo un tema con los .jpg que no me los acepta. Falta revisar qué pasa con la validación de los .jpg acá y en los demás formularios que lo requieran

SoraBeat committed:
Validacion de imagen arreglada

milandre commented on src/Components/common/Form/InputImage.jsx:
```diff
@@ -0,0 +1,70 @@
+	const inputImage = useRef();
+	const [fileName, setFileName] = useState("");
+
+	const fileTypes = [
```
Acá faltaría verificar de igual forma la validación, como lo comentado en #1

SoraBeat committed:
Mover declaracion de tipos de archivos

milandre reviewed (approved):
@SoraBeat aprobado y listo para hacerle merge. Hay un tema con los archivos .jpg que no me deja agregar y me salta el cartel de "Selecciona una imagen jpg o png". Estaría bueno revisar eso antes de hacerle merge

DiegoSantelices10 commented:
lauta te dejo una solución que me funciono con el tema de la validación de la imágenes. por las!

![Screenshot_13](https://user-images.githubusercontent.com/70447941/200460511-25d894b1-54bf-4867-bf1a-2052597ce1a8.png)

SoraBeat committed:
Error al importar imagen

Se tuvo que cambiar el metodo de validacion ya que estaba mal planteado y agregar el formato jpeg ya que algunas imagenes jpg los toma con ese formato

SoraBeat committed:
Borrado de console log inecesario xd

SoraBeat committed:
Pulleo desde dev y cambios de estilos formulario miembros
